### PR TITLE
Make sure the `sourcemaps` build option is respected, even when providing a custom `terser` config

### LINKED ISSUE #306: Ember cli sourcemaps setting not respected by 4.0.2?
Unlike the documentation claims, ember-cli-terser seems to generate sourcemaps even for production builds regardless of how the ember-cli sourcemaps setting is configured.

Only way to disable sourcemaps from what I can tell is explicitly set `'ember-cli-terser'.terser.sourceMap = false`

### PR DESCRIPTION
**Pull request by bertdeblock**
Closes https://github.com/ember-cli/ember-cli-terser/issues/306.
Closes https://github.com/ember-cli/ember-cli-terser/pull/308.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -39,6 +35,14 @@ module.exports = {
     }
 
     this._terserOptions = Object.assign({}, defaultOptions, addonOptions);
+
+    if (
```
Comment (bertdeblock) [inline index.js]: Executing this check after the `Object.assign` logic makes sure the `sourcemaps` option is still respected, even when the user provides their own `terser` config.

File: index.js [hunk 2]
```diff
@@ -39,6 +35,14 @@ module.exports = {
...
+      app.options.sourcemaps &&
+      this._sourceMapsEnabled(app.options.sourcemaps) === false &&
+      typeof this._terserOptions.terser.sourceMap === 'undefined'
```
Comment (bertdeblock) [inline index.js]: In case the user already provides the `sourceMap` option.

**Comment by bertdeblock** [comment]
Made https://github.com/ember-cli/ember-cli-terser/pull/314 as a prerequisite, so we can merge stuff like this with more confidence.

**Review by kategengler — APPROVED** [review]

**Comment by bertdeblock** [comment]
@kategengler I updated this branch using the work that landed in #314.

**Pull request merged**